bin/conform-templates.php

```diff
@@ -0,0 +1,299 @@
+    }
+    $content = preg_replace(
+        '/\/phpcs\.xml export-ignore/',
+        '/phpbench.json export-ignore' . PHP_EOL . '/phpcs.xml export-ignore',
+        $content
+    );
+}
+if (file_exists('.docheader')) {
+    if (isset($composer['require-dev']['malukenho/docheader'])) {
+        $content = preg_replace(
+            '/\/\.coveralls.yml export-ignore/',
+            '/.coveralls.yml export-ignore' . PHP_EOL . '/.docheader export-ignore',
+            $content
+        );
+    } else {
+        unlink('.docheader');
+    }
+}
+file_put_contents('.gitattributes', $content);
+
+// .gitignore
+$content = file_get_contents(__DIR__ . '/../template/.gitignore');
+if (! $hasDocs) {
+    $content = preg_replace('/^docs\/html\/\s+/m', '', $content);
+    $content = preg_replace('/^zf-mkdoc-theme\/\s+/m', '', $content);
+    $content = preg_replace('/^zf-mkdoc-theme\.tgz\s+/m', '', $content);
+}
+file_put_contents('.gitignore', $content);
+
+// .travis.yml - create only when does not exist
+if (! file_exists('.travis.yml')) {
+    copy(__DIR__ . '/../template/.travis.yml', '.travis.yml');
+}
+
+// composer.json
+// - checks repository description
+// - checks order of sections
+// - removes default type library
+// - updates scripts
+// - updates license
+// - updates support links
+// - updates "config"
+// - removes "minimum-stability" and "prefer-stable"
+// - checks keywords
+// @todo: check branch-alias
+
+$templateContent = json_decode(file_get_contents(__DIR__ . '/../template/composer.json'), true);
+
+$sectionOrder = [];
+foreach ($templateContent as $section => $value) {
+    $sectionOrder[] = $section;
+}
+
+$content = $composer;
+if (isset($content['type']) && $content['type'] === 'library') {
+    unset($content['type']);
+}
+$content['license'] = $templateContent['license'];
+$content['support'] = $templateContent['support'];
+if (! $hasDocs) {
+    unset($content['support']['docs']);
+}
+foreach ($content['support'] as &$supportLink) {
+    $supportLink = strtr($supportLink, $replace);
+}
+
+$content['config'] = $templateContent['config'];
+$content['scripts'] = $templateContent['scripts'];
+
+if (empty($content['keywords'])) {
+    fwrite(STDERR, 'Missing "keywords" in composer.json');
+} else {
+    $hasZf = false;
+    $hasZendframework = false;
+    foreach ($content['keywords'] as &$keyword) {
+        if ($keyword === 'zf2') {
+            $keyword = 'zf';
+            $hasZf = true;
+        } elseif ($keyword === 'zf') {
+            $hasZf = 'zf';
+        } elseif ($keyword === 'zendframework') {
+            $hasZendframework = true;
+        }
+    }
+
+    if (! $hasZendframework) {
+        array_unshift($content['keywords'], 'zendframework');
+    }
+    if (! $hasZf) {
+        array_unshift($content['keywords'], 'zf');
+    }
+}
+
+if (file_exists('.docheader')) {
+    array_unshift($content['scripts']['check'], '@license-check');
+    $content['scripts']['license-check'] = 'docheader check src/ test/';
+}
```

Comment:
These two operations should only be done _if those elements are not already defined_.

Reply:
I override all scripts before to match these from templates - see line 179, and these are not defined. Anyway, I think we are going to remove it soon - and use licence check from zend-coding-standards library.

Reply:
Exactly. :)